crates/runtime-table-partition/src/creator/filename.rs

```diff
@@ -0,0 +1,321 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! Create filenames from [`ScalarValue`]s and infer [`ScalarValue`]s from
+//! filenames.
+//!
+//! `serde_qs` crate is used to serialize/deserialize [`ScalarValue`]s into
+//! [`String`]s that are URL encoded. This is helpful because we get the
+//! encoding and decoding functionality and special characters are escaped
+//! making them filesystem compatible.
+//!
+//! A subset of the [`ScalarValue`] variants are copied into a new type
+//! [`SupportedScalarValue`] so that we can derive `Serialize` and `Deserialize`
+//! on variants that have types that can be serialized/deserialized.
+
+use datafusion::scalar::ScalarValue;
+use serde::{Deserialize, Serialize};
+use snafu::prelude::*;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub))]
```

Comment:
What does this do?

Reply:
It makes the generated *Snafu types public. It's unnecessary with how my code ended up. I'll remove unless/until needed.